tests/lib/rules/no-direct-mutation-state.js

```diff
@@ -70,6 +70,14 @@ ruleTester.run('no-direct-mutation-state', rule, {
       '}'
     ].join('\n')
   }, {
+    code: [
+      'class Hello extends React.Component {',
+      '  constructor() {',
+      '    this.state.foo = 1;',
```

Comment:
This has been my favorite find in our codebase where we were assigning to state.

```
this.setState({
  foo: this.state.foo++
});
```